ServiceFinder/ServiceFinder.BLL/Services/GenericService.cs

```diff
@@ -40,6 +42,7 @@ public async virtual Task<TModel> UpdateAsync(Guid id, TModel model, Cancellatio
 
         public async virtual Task DeleteAsync(Guid id, CancellationToken cancellationToken)
         {
+            if (!await CheckIfEntityExists(id, cancellationToken)) throw new ModelNotFoundException(id);
             var entity = await _repository.GetByIdAsync(id, cancellationToken);
```

Comment:
you do double query into your database. the first one in CheckIfEntityExists, and the second one after it. lol

Reply:
you do double query into your database. the first one in CheckIfEntityExists, and the second one after it. lol #2